### PR TITLE
[Utils.DOM] - Clean dom utils

### PR DESCRIPTION
**Pull request by bluong**
Last minute cleanup of the DOM utils module.

API Changes:
* `Utils.DOM.POLYFILL_TIMEOUT_MILLISECONDS` has been renamed to `Utils.DOM.SCREEN_REFRESH_RATE_MILLISECONDS`
* `Utils.DOM.boxesOverlap()` has been renamed to `Utils.DOM.clientRectsOverlap()`
* `Utils.DOM.boxIsInside()` has been renamed to `Utils.DOM.clientRectInside()`

Link to #2272

### CODE REVIEW

File: src/utils/domUtils.ts [hunk 1]
```diff
@@ -20,13 +20,13 @@ export module Utils {
       return bbox;
     }
 
-    export var POLYFILL_TIMEOUT_MILLISECONDS = 1000 / 60; // 60 fps
+    export var REQUEST_ANIMATION_FRAME_TIMEOUT_MILLISECONDS = 1000 / 60; // 60 fps
```
Comment (aicioara) [inline src/utils/domUtils.ts]: Can we shorten it in any way?

Comment (aicioara) [inline src/utils/domUtils.ts]: maybe drop MILLISECONDS?

File: src/utils/domUtils.ts [hunk 2]
```diff
@@ -20,13 +20,13 @@ export module Utils {
...
 
     export function requestAnimationFramePolyfill(fn: () => void) {
```
Comment (aicioara) [inline src/utils/domUtils.ts]: What is a `Polyfill` do you think this is an appropriate name now that the constant is named differently?

File: src/utils/domUtils.ts (hunk 1 above)
Comment (aicioara) [inline src/utils/domUtils.ts]: We were arguing this should not be exported? I could not find a way not to export it, unless actually using the `requestAnimationFramePolyfill` itself

**Comment by aicioara** [comment]
Links here https://github.com/palantir/plottable/issues/2272

File: src/utils/domUtils.ts (hunk 2 above)
Comment (jtlan) [inline src/utils/domUtils.ts]: I think keeping the old name is better, since the *actual* `requestAnimationFrame` timeout could be different.

File: src/utils/domUtils.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export module Utils {
       return selection;
     }
 
-    export function boxesOverlap(boxA: ClientRect, boxB: ClientRect) {
+    export function clientRectOverlap(boxA: ClientRect, boxB: ClientRect) {
```
Comment (jtlan) [inline src/utils/domUtils.ts]: `clientRectsOverlap`?

File: src/utils/domUtils.ts (hunk 2 above)
Comment (bluong) [inline src/utils/domUtils.ts]: Based on the first google result of "polyfill" you get...

"A polyfill, or polyfiller, is a piece of code (or plugin) that provides the technology that you, the developer, expect the browser to provide natively." - https://remysharp.com/2010/10/08/what-is-a-polyfill

Which basically says that the word polyfill here means almost nothing.  Our `Map`s by this definition are I think polyfills and thus having the variable name `POLYFILL_TIMEOUT_MILLISECONDS` I think is really not useful

File: src/utils/domUtils.ts (hunk 1 above)
Comment (bluong) [inline src/utils/domUtils.ts]: `RenderPolicies.Timeout` uses this value as the default value, and that mainly is the thing that's blocking from hiding this variable.  After thinking about it, I think it's ok to expose this variable

File: src/utils/domUtils.ts (hunk 3 above)
Comment (bluong) [inline src/utils/domUtils.ts]: I went for this approach since I was thinking "does clientRect 1 overlap clientRect 2?".  That said, your proposal works just as well.

What do you think?

Comment (jtlan) [inline src/utils/domUtils.ts]: `clientRectsOverlap()` makes more sense to me. Also avoids suggesting that it returns the overlap.

File: src/utils/domUtils.ts (hunk 2 above)
Comment (jtlan) [inline src/utils/domUtils.ts]: >Which basically says that the word polyfill here means almost nothing.  Our `Map`s by this definition are I think polyfills and thus having the variable name `POLYFILL_TIMEOUT_MILLISECONDS` I think is really not useful

I don't see how the first part follows from the second part here.

For that matter, is there a reason this needs to be exported?

Comment (bluong) [inline src/utils/domUtils.ts]: `POLYFILL_TIMEOUT_MILLISECONDS` can be read as the "timeout for a polyfill in milliseconds".  However, this isn't the timeout for just a polyfill, it is the timeout for the requestAnimationFrame polyfill.

Comment (bluong) [inline src/utils/domUtils.ts]: It needs to be exported since the `Timeout` RenderPolicy uses it as the default timeout

**Comment by aicioara** [comment]
Suggestion `REQUEST_ANIMATION_FRAME_TIMEOUT_MILLISECONDS` -> `ANIMATION_FRAME_TIMEOUT_MILLISECONDS`. No verb.

**Comment by bluong** [comment]
Renames. NOQE

**Comment by jtlan** [comment]
Docs?

File: src/utils/domUtils.ts [hunk 4]
```diff
@@ -64,30 +100,50 @@ export module Utils {
...
 
-    export function boxIsInside(inner: ClientRect, outer: ClientRect) {
+    /**
+     * Checks if the first ClientRect is inside the second.
+     * 
+     * @param {ClientRect} clientRectA The first ClientRect
+     * @param {ClientRect} clientRectB The second ClientRect
+     * @returns {boolean} If the first ClientRect is inside the second.
+     */
+    export function clientRectInside(clientRectA: ClientRect, clientRectB: ClientRect) {
```
Comment (jtlan) [inline src/utils/domUtils.ts]: Why did you rename these from `inner` and `outer`? I think the old names were clearer.

**Comment by jtlan** [comment]
[-1] param names seem less clear now.

File: src/utils/domUtils.ts (hunk 4 above)
Comment (bluong) [inline src/utils/domUtils.ts]: Mainly to reflect the docs.  The `inner` one is not really confirmed to be in the second one unless the function returns true.

Do you think it should be `innerClientRect`?

Comment (jtlan) [inline src/utils/domUtils.ts]: No, because people using Typescript should have tooling available to help them out there.

Javascript consumers would probably be looking at the docs or the d.ts file, otherwise the parameter name wouldn't be available to help them anyway.
